### PR TITLE
new package: py-reproject

### PR DESCRIPTION
**Pull request by aweits**
add setuptools build/run dep to py-astropy-healpix

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-extension-helpers/package.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+class PyExtensionHelpers(PythonPackage):
+    """The extension-helpers package includes convenience helpers to
+    assist with building Python packages with compiled C/Cython
+    extensions. It is developed by the Astropy project but is intended
+    to be general and usable by any Python package."""
+
+    homepage = 'https://github.com/astropy/astropy-helpers'
+    url = 'https://pypi.io/packages/source/e/extension-helpers/extension-helpers-0.1.tar.gz'
+
+    depends_on('py-setuptools', type='build')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-extension-helpers/package.py]:
```suggestion
    depends_on('py-setuptools@30.3:', type='build')
```

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-extension-helpers/package.py]: Also add:
```python
depends_on('python@3.6:', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-reproject/package.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    reproject to/from HEALPIX projections by relying on the
+    astropy-healpix package."""
+
+    homepage = 'https://reproject.readthedocs.io/'
+    url = 'https://pypi.io/packages/source/r/reproject/reproject-0.7.1.tar.gz'
+
+    depends_on('py-setuptools', type='build')
+    depends_on('py-cython', type='build')
+    depends_on('py-extension-helpers', type='build')
+    depends_on('python@3.5:', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-reproject/package.py]:
```suggestion
    depends_on('python@3.6:', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-reproject/package.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    pixels on the celestial sphere (like Montage). It can also
+    reproject to/from HEALPIX projections by relying on the
+    astropy-healpix package."""
+
+    homepage = 'https://reproject.readthedocs.io/'
+    url = 'https://pypi.io/packages/source/r/reproject/reproject-0.7.1.tar.gz'
+
+    depends_on('py-setuptools', type='build')
+    depends_on('py-cython', type='build')
+    depends_on('py-extension-helpers', type='build')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-reproject/package.py:28]: Missing a build dep on `py-setuptools-scm`

File: var/spack/repos/builtin/packages/py-reproject/package.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    depends_on('python@3.5:', type=('build', 'run'))
+    depends_on('py-numpy@1.13:', type=('build', 'run'))
+    depends_on('py-astropy@3.2:', type=('build', 'run'))
+    depends_on('py-scipy@0.9:', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-reproject/package.py]:
```suggestion
    depends_on('py-scipy@1.1:', type=('build', 'run'))
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]
I usually put versions above dependencies.

File: var/spack/repos/builtin/packages/py-extension-helpers/package.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    to be general and usable by any Python package."""
+
+    homepage = 'https://github.com/astropy/astropy-helpers'
+    url = 'https://pypi.io/packages/source/e/extension-helpers/extension-helpers-0.1.tar.gz'
+
+    version('0.1', sha256='ac8a6fe91c6d98986a51a9f08ca0c7945f8fd70d95b662ced4040ae5eb973882')
+
+    depends_on('python@3.6:', type=('build', 'run'))
+    depends_on('py-setuptools@30.3:', type='build')
+    depends_on('py-setuptools', type='build')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-extension-helpers/package.py]: You can remove this line, it duplicates the previous line

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-extension-helpers/package.py (hunk 5 above)
Comment (aweits) [reply var/spack/repos/builtin/packages/py-extension-helpers/package.py]: uggh, perhaps the coffee isn't strong enough...

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**